People in place and in motion full implementation, template for acoustic profile test, and various other changes.

Zoltorion opened the pull request:
- JsonToString mixin added to db_schema_classes and used with most Test subclasses and some others.
- SectionCutterTest and Section classes refactored to comply with JsonToString and consequently be more similar to other implementations.
- Removed a couple extension methods from firestore_functions that were unused and useless.
- Added new class in theme.dart ChipLabelColor.
- Refactored spatial_boundaries_test a bit.
- Probably something else I forgot idk there's a lot here.

d-minic commented:
Changes look good. A few things: I'm sure we've already considered this, but when we get timers implemented correctly we should probably make it so users can only start placing points/lines when the timer begins. And the second, bigger thing is standing points. The people tests and the acoustics tests all user standing points, for the people tests it should just involve rendering them on the map and not doing anything further w/ them.

Zoltorion commented:
> And the second, bigger thing is standing points. The people tests and the acoustics tests all user standing points, for the people tests it should just involve rendering them on the map and not doing anything further w/ them.

Wow I somehow completely forgot about the standing points, although I also wasn't quite sure what needs to be done with them on the people tests either. I'll go back and add those in another commit real quick if that's all that needs to be done with them.

michaelsoffline reviewed (approved):
Everything looks like it works fine on iOS. There's some visual adjustments I want to make but I'll save those for after this gets merged.

michaelsoffline commented:
> Oh also, as for the frontend we probably want these tests to be put into a SafeArea as some of the content is cut off by certain areas on my phone.

I found a solution online that I want to try that would basically conditionally use a Safe Area in the test depending on whether the platform is Android or iOS. Flutter has a way of tracking what platform you're on so it should resolve this issue. The Safe Area seems to be necessary for Android because of the way the status bar works on there but iOS doesn't have any of those constraints and allows the screen to blend with the status bar without issue. The Safe Area just makes it look like I'm running an older app that hasn't been updated to accommodate the iPhones with larger screen sizes.

Zoltorion commented:
> > Oh also, as for the frontend we probably want these tests to be put into a SafeArea as some of the content is cut off by certain areas on my phone.
> 
> I found a solution online that I want to try that would basically conditionally use a Safe Area in the test depending on whether the platform is Android or iOS. Flutter has a way of tracking what platform you're on so it should resolve this issue. The Safe Area seems to be necessary for Android because of the way the status bar works on there but iOS doesn't have any of those constraints and allows the screen to blend with the status bar without issue. The Safe Area just makes it look like I'm running an older app that hasn't been updated to accommodate the iPhones with larger screen sizes.

Yeah that sounds like a nice change if you can get it working but also is a very minor issue from the sound of it.

Zoltorion commented:
Updated list of significant changes:
- New class for StandingPoint
- Static typing added to pretty much everywhere that previously was using a dynamic List or some similarly unsafe type when working with standing points
  - Also similar static typing added to polygonPoints field of Project class, now List<LatLng> rather than just List and changes made in many other places as well since previously it was used with the assumption of it being GeoPoint List
- Changed Sets holding collectionIDs of tests that use standing points or timer to static maps in Test similar to the 4 other ones we already had there. This means that now each class that uses those needs to register to the appropriate set in addition to the 4 Maps that all Test subclasses need to do. Additionally, Test class will now have a method for each of those Sets that returns whether or not the given collectionID is part of that test type Set. Currently this is only isStandingPointTest but the intent is to have a timer equivalent as needed once that is started.
- Multiple cool widgets added to widgets.dart that are used in both people tests but could be adapted to others if we transition them to that style.

d-minic commented on lib/create_test_form.dart:
But this isn't urgent or anything since List can't be null it just throws the error, but doesn't break anything.

michaelsoffline reviewed (approved):
Still looks good on iOS. Merge when you're ready.
